build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -70,7 +71,7 @@ public final class CompilationDatabaseGenerator {
 	 * Checked on each build
 	 * Used before we look up the environment
 	 */
-	private Map<ITool, String> toolMap = new HashMap<>();
+	private Map<String, String> toolMap = new HashMap<>();
```

Comment:
Because we now have a key of a string it is much less obvious what that key is, so please document it. Perhaps a line like this in the javadoc: `Map of compiler name (as calculated by {@link #getCompilerName(String)}) to the absolute path of the compiler.`